docs/key-concepts.md

```diff
@@ -0,0 +1,122 @@
+
+These logs are essential for troubleshooting problems, auditing system activity, and understanding user interactions. In software development and operations, logs are generated by various components, including applications, servers, databases, and networking devices.
+
+This is what a sample log data in a system would look like.
+
+| Timestamp            | Event Description                       | Severity | Contextual Data                                    |
+| -------------------- | --------------------------------------- | -------- | -------------------------------------------------- |
+| 2024-02-11T08:35:00Z | Application server started successfully | INFO     | Server: app-01, Environment: Production            |
+| 2024-02-11T09:12:45Z | Disk usage exceeded threshold           | WARNING  | Server: db-03, Disk Usage: 85%, Threshold: 80%     |
+| 2024-02-11T10:20:30Z | Critical error in database connection   | ERROR    | Server: db-02, Error Code: 500, Connection Timeout |
+| 2024-02-11T11:05:15Z | User authentication failed              | ERROR    | Username: johndoe, IP: 192.168.1.100               |
+| 2024-02-11T12:30:00Z | Scheduled job completed successfully    | INFO     | Job: Backup, Duration: 00:45:00                    |
+
+### Log ingestion
+
+As the name suggests, it refers to the process of **collecting and importing log data** from various sources into a centralized system for analysis and storage. It involves capturing log events generated by applications, servers, and other components, parsing and enriching the log data, and forwarding it to a centralized logging system or storage backend. Log ingestion systems often support various protocols and formats for data collection, including syslog, JSON, and structured logging.
+
+## 2. Metrics 📊
+
+Metrics are quantitative measurements that provide insight into the **performance** and **health** of a system. Unlike logs, which provide detailed event-based information, metrics focus on numerical data points over time. Common metrics include **CPU usage**, **memory utilization**, **response times**, **throughput**, and **error rates**.
+
+Metrics play a vital role in overseeing system performance, pinpointing irregularities, and uncovering patterns that could signify either emerging problems or chances for optimization. These metrics are usually gathered and consolidated from diverse origins within the system, including applications, servers, and infrastructure components.
+
+## 3. Traces 🔍
+
+Traces are a record of the execution path of a request as it moves through a distributed system. They provide detailed information about each step taken by the request, including timestamps, service interactions, and any errors encountered along the way. 
+
```

Comment:
I think it would be worth it to attach an example and screenshot here about traces. We can talk about a  2-3 functions working together, and the tracing data for that is time taken to go from Func A to Func B to Func C. Feel free to include a different example which might be easier to understand